### DISCUSSION TITLE
Turbulence driving broken - Decaying turbulence

### DISCUSSION THREAD

**Discussion by saisandeepdammati** [PrincetonUniversity/athena#525] [Q&A]
Howdy!

I am using athena++ from the master brach with latest git commit **ca8ba578e085fe37a2962db7f5bff8d35be380a6** and trying to run a simple decaying turbulence problem (using src/pgen/turb.cpp) in a periodic cube of size 0.1125 cm on a 128^3 uniform grid on my local macbook on single core with one meshblock. I see that the code runs for 38 cycles and the then time-step drops drastically to very small value (1.e-22) and the run breaks. The only changes I made in turb.cpp is that I read the value of density and pressure from the <problem> of the athinput file as shown below in `MeshBlock::ProblemGenerator(ParameterInput *pin)`  function. In history output, I can see that after cycle 38 suddenly total energy in the domain blows up. I am not sure what is the source of the error. 

The configuration script and athena configuration for the run are shown below. I am also attaching the job file, the history output file and stdout as .txt files.

**Any input is highly appreciated.**

------------------ CHANGE IN TURB.CPP ----------------------------------
`phydro->u(IDN,k,j,i) = pin->GetReal("problem","d");`
`if (NON_BAROTROPIC_EOS) {
            phydro->u(IEN,k,j,i) = pin->GetReal("problem","pr")/(peos->GetGamma()-1);`

------------------ CONFIGURATION SCRIPT ----------------------------------
python3 configure.py --prob=turb -mpi --eos=adiabatic --flux=hllc --nghost=2 -hdf5 -fft --fftw_path=/usr/local --hdf5_path=/usr/local/hdf5-install

------------------ ATHENA CONFIGURATION ----------------------------------

Your Athena++ distribution has now been configured with the following options:
  Problem generator:          turb
  Coordinate system:          cartesian
  Equation of state:          adiabatic
  Riemann solver:             hllc
  Magnetic fields:            OFF
  Number of scalars:          0
  Special relativity:         OFF
  General relativity:         OFF
  Frame transformations:      OFF
  Self-Gravity:               OFF
  Super-Time-Stepping:        OFF
  Debug flags:                OFF
  Code coverage flags:        OFF
  Linker flags:                -L/usr/local/lib -L/usr/local/hdf5-install/lib  -lfftw3 -lhdf5
  Floating-point precision:   double
  Number of ghost cells:      2
  MPI parallelism:            ON
  OpenMP parallelism:         OFF
  FFT:                        ON
  HDF5 output:                ON
  HDF5 precision:             single
  Compiler:                   g++
  Compilation command:        mpicxx  -I/usr/local/include -I/usr/local/hdf5-install/include -O3 -std=c++11

[hit.decaying.job.txt](https://github.com/PrincetonUniversity/athena/files/11858368/hit.decaying.job.txt)
[HIT.n128.Decaying.vl2.hst.txt](https://github.com/PrincetonUniversity/athena/files/11858369/HIT.n128.Decaying.vl2.hst.txt)
[HIT.n128.Decaying.vl2.out.txt](https://github.com/PrincetonUniversity/athena/files/11858375/HIT.n128.Decaying.vl2.out.txt)

**Comment by c-white** [comment]
I see the following big changes from the standard problem:
- The CFL number has been raised from 0.3 to 0.45.
- The adiabatic index is about 1.2 rather than 5/3.
- Instead of density and pressure both being 1, they are now about 1e-3 and 1e6, so the temperature has changed from 1 to 9e8. If this is trying to make the code use CGS internally, then I'll warn that this is usually a bad idea. There are hard-coded small and large values used in traps for bad numerical behavior all throughout the code, so it's best to scale the problem to have order-unity numbers being evolved.
- The problem is 128^3 instead of 32^3.
- `nlow`, `nhigh`, and `expo` are no longer specified. Someone more familiar with the turbulence module can comment, but I can only imagine this results in driving all the way down to the grid scale.

**Comment by saisandeepdammati** [comment]
Hi Chris,

Thanks for your input. I figured that it is an issue of numerical stability. I repeated the same run with a CFL of 0.3 and it works fine.
